### PR TITLE
Move PyPi checks to twine

### LINKED ISSUE #1023: move to twine check
see https://github.com/pypa/readme_renderer/issues/118

### PR DESCRIPTION
**Pull request by gaborbernat**
Resolves #1023.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -70,9 +70,6 @@ def main():
             "console_scripts": ["tox=tox:cmdline", "tox-quickstart=tox._quickstart:main"]
         },
         python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*",
-        setup_requires=[
-            "setuptools-scm"
```
Comment (anthrotype) [inline setup.py]: If you remove setuptools-scm from setup_requires the setup.py won’t work with pip versions that don’t understand the pyproject.toml build requirements, no?

Comment (gaborbernat) [reply setup.py]: Not in this case, as setuptools-scm now is a source distribution to sdist dependency. It does not get used when we build from a source distribution a wheel, or when we install that. If you look into the output the sdist command you'll see the generated PKG-INFO already contains the version info. Furthermore, in our code base at no point do we use setuptools-scm (instead we generate the version.py https://github.com/tox-dev/tox/blob/master/setup.py#L52 at sdist time, and refer to that). 

```bash
Metadata-Version: 2.1
Name: tox
Version: 3.4.1.dev21+gf2d6c71
Summary: virtualenv-based automation of test activities
Home-page: https://tox.readthedocs.org/
Author: holger krekel
Author-email: holger@merlinux.eu
License: https://opensource.org/licenses/MIT
```

**Comment by anthrotype** [comment]
I see. But then pip installing from a local git checkout will depend on a relatively recent version of pip, that can bootstrap the build requirements.

**Comment by gaborbernat** [comment]
Installing tox HEAD (aka a git checkout) with an old pip? Seems a niche enough to ignore it. The only officially supported method of installing tox is via pip and PyPi. And for that setuptools-scm is not needed.

**Comment by anthrotype** [comment]
Also python setup.py install would probably break, but I presume easy_install is de facto deprecated these days.
I’m interested because I may do the same in other projects of mine (remove setuptools-scm from setup_requires).

**Pull request merged**